result4k/core/src/main/kotlin/dev/forkhandles/result4k/iterables.kt

```diff
@@ -18,3 +18,42 @@ fun <T, E> Iterable<Result<T, E>>.partition(): Pair<List<T>, List<E>> {
     }
     return Pair(oks, errs)
 }
+
+// Traverse family of functions
+fun <T, Tʹ, E> Iterable<T>.foldResult(
+    initial: Result<Tʹ, E>,
```

Comment:
Is there a reason for this to be a Result instead of an E?  Is there any time you would not pass Success(e) as this parameter?

Reply:
well, not intentionally, but if the initial is coming from chain of result operations, it's nicer to fold it rather than map and then fold internally.

Reply:
I'm not super opinionated, but that's the reason, I can change it.